publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -111,8 +111,10 @@ public ResponseEntity send(@ApiParam(value = "message protocol", required = true
                 return new ResponseEntity(e.getMessage(), HttpStatus.NOT_FOUND);
             }
         }
-        SendResult result = messageService.send(body, msgService, userDomain, tag, routingKey);
-        return new ResponseEntity(result, messageService.getHttpStatus());
+        synchronized(this) {
+            SendResult result = messageService.send(body, msgService, userDomain, tag, routingKey);
+            return new ResponseEntity(result, messageService.getHttpStatus());
+        }
```

Comment:
Isn't the real problem that [MessageServiceRMQImpl.java](https://github.com/eiffel-community/eiffel-remrem-publish/blob/6668b3baff7d92f4ca6c381716fbce2826326bca/publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/service/MessageServiceRMQImpl.java) isn't thread-safe? If so this is just band-aid that treats the symptom rather than the cause.

Reply:
I will raise a new PR for virtualHost change. 
I will go through the MessageServiceRMQImpl.java and will check regarding how can we make the complete appication thread-safe.